lib/linguist/heuristics.yml

```diff
@@ -157,6 +157,19 @@ disambiguations:
     pattern: '# Microsoft Developer Studio Generated Build File'
   - language: Faust
     pattern: '\bprocess\s*[(=]|\b(library|import)\s*\(\s*"|\bdeclare\s+(name|version|author|copyright|license)\s+"'
+- extensions: ['.e']
+  rules:
+  - language: E
+    pattern:
+    - '^\s*(def|var)\s+(.+):='
+    - '^\s*(def|to)\s+(\w+)(\(.+\))?\s+{'
+    - '^\s*(when)\s+(\(.+\))\s+->\s+{'
+  - language: Eiffel
+    pattern:
+    - '^\s*{/\w+'
+    - '^\s*(across|elseif|feature|from|inherit|inspect|note|when)\s+'
+  - language: Euphoria
+    named_pattern: euphoria
```

Comment:
Do we really need a named pattern? The patterns are only being referenced once, and they aren't hairy enough to warrant burying at the bottom of the file.

Reply:
I used a named pattern here because **Euphoria** is referenced in extensions `['.e']` and `['.ex']`. Is that appropriate?

Reply:
Ah, so it is. 🤔 Never mind.